Assets/Tests/Runtime/NetworkedMessageChannelTests.cs

```diff
@@ -0,0 +1,263 @@
+                m_ClientScopes[i].BindInstanceAsSingle(m_ClientNetworkManagers[i]);
+                m_ClientScopes[i].FinalizeScopeConstruction();
+            }
+
+            m_ServerScope = new DIScope();
+            m_ServerScope.BindInstanceAsSingle(m_ServerNetworkManager);
+            m_ServerScope.FinalizeScopeConstruction();
+
+            base.OnServerAndClientsCreated();
+        }
+
+        protected override IEnumerator OnTearDown()
+        {
+            m_Subscriptions.Dispose();
+            for (int i = 0; i < NumberOfClients; i++)
+            {
+                m_ClientScopes[i] = new DIScope();
+                m_ClientScopes[i].BindInstanceAsSingle(m_ClientNetworkManagers[i]);
+                m_ClientScopes[i].FinalizeScopeConstruction();
+            }
+            m_ServerScope.Dispose();
+            return base.OnTearDown();
+        }
+
+        void InitializeNetworkedMessageChannels<T>(int nbClients, int nbSubscribers, T expectedValue, out NetworkedMessageChannel<T>[] emptyMessageChannelClients, out NetworkedMessageChannel<T> emptyMessageChannelServer) where T : unmanaged, INetworkSerializeByMemcpy
+        {
+            emptyMessageChannelClients = new NetworkedMessageChannel<T>[nbClients];
+            for (int i = 0; i < nbClients; i++)
+            {
+                emptyMessageChannelClients[i] = new NetworkedMessageChannel<T>();
+                m_ClientScopes[i].InjectIn(emptyMessageChannelClients[i]);
+            }
+
+            emptyMessageChannelServer = new NetworkedMessageChannel<T>();
+            m_ServerScope.InjectIn(emptyMessageChannelServer);
+
+            m_Subscriptions = new DisposableGroup();
+            for (int i = 0; i < nbClients; i++)
+            {
+                for (int j = 0; j < nbSubscribers; j++)
+                {
+                    var numClient = i;
+                    var numSub = j;
+                    m_Subscriptions.Add(emptyMessageChannelClients[i].Subscribe(message =>
+                    {
+                        Debug.Log($"Received message on client {numClient} in subscription {numSub}.");
+                        m_NbMessagesReceived++;
+                        Assert.AreEqual(message, expectedValue);
+                    }));
+                }
+            }
+
+            for (int j = 0; j < nbSubscribers; j++)
+            {
+                var numSub = j;
+                m_Subscriptions.Add(emptyMessageChannelServer.Subscribe(message =>
+                {
+                    Debug.Log($"Received message on server in subscription {numSub}.");
+                    m_NbMessagesReceived++;
+                    Assert.AreEqual(message, expectedValue);
+                }));
+            }
+        }
+
+        [UnityTest]
+        public IEnumerator EmptyNetworkedMessageIsReceivedByAllSubscribersOnAllClientsAndServer([ValueSource(nameof(s_NbClients))] int nbClients, [ValueSource(nameof(s_NbSubs))] int nbSubscribers)
```

Comment:
for something as simple as a list of ints, you can use Values
https://docs.nunit.org/articles/nunit/writing-tests/attributes/values.html
ValueSource is mostly for runtime generated values